tests/attr/test_feature_ablation.py

```diff
@@ -66,6 +66,48 @@ def test_multi_sample_ablation_with_mask(self):
             ablations_per_eval=(1, 2, 3),
         )
 
+    def test_single_sample_ablation_with_mask_scalar(self):
+        net = BasicModel_MultiLayer()
+        inp = torch.tensor([[2.0, 10.0, 3.0]], requires_grad=True)
+        mask = torch.tensor([[0, 0, 1]])
+
+        # testing int, float, 0d and 1d tensor output
+        for func in [
+            lambda inp: torch.sum(net(inp)).item(),
+            lambda inp: torch.sum(net(inp)),
+            lambda inp: torch.sum(net(inp)).reshape(1),
+            lambda inp: int(torch.sum(net(inp)).item()),
+        ]:
+            self._ablation_test_assert(
+                func,
+                inp,
+                [[82.0, 82.0, 24.0]],
+                feature_mask=mask,
+                ablations_per_eval=(1,),
+                target=None,
+            )
+
+    def test_multi_sample_ablation_with_mask_scalar(self):
+        net = BasicModel_MultiLayer()
+        inp = torch.tensor([[2.0, 10.0, 3.0], [20.0, 50.0, 30.0]], requires_grad=True)
+        mask = torch.tensor([[0, 0, 1]])
+
+        # testing int, float, 0d and 1d tensor output
+        for func in [
+            lambda inp: torch.sum(net(inp)).item(),
+            lambda inp: torch.sum(net(inp)),
+            lambda inp: torch.sum(net(inp)).reshape(1),
+            lambda inp: int(torch.sum(net(inp)).item()),
```

Comment:
nit: perhaps we can split this into four tests ? This will reduce test-wise execution time and will help to narrow down the failing case.